optaplanner-persistence/optaplanner-persistence-jsonb/src/test/java/org/optaplanner/persistence/jsonb/api/score/AbstractScoreJsonbAdapterTest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.persistence.jsonb.api.score;
+
+import java.io.Serializable;
+
+import javax.json.bind.Jsonb;
+import javax.json.bind.JsonbBuilder;
+import javax.json.bind.JsonbException;
+
+import org.optaplanner.core.api.score.Score;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+public class AbstractScoreJsonbAdapterTest {
+
+    // ************************************************************************
+    // Helper methods
+    // ************************************************************************
+
+    protected <S extends Score, W extends TestScoreWrapper<S>> void assertSerializeAndDeserialize(S expectedScore, W input) {
+        String jsonString;
+        W output;
+        try {
+            Jsonb jsonb = JsonbBuilder.create();
+            jsonString = jsonb.toJson(input);
+            output = (W) jsonb.fromJson(jsonString, input.getClass());
+        } catch (JsonbException e) {
+            throw new IllegalStateException("Marshalling or unmarshalling for input (" + input + ") failed.", e);
+        }
+        assertEquals(expectedScore, output.getScore());
+
+        String newLine = System.lineSeparator();
+        String regex;
+        if (expectedScore != null) {
+            regex = "\\{\"score\":\"" // Start of element
+                    + expectedScore.toString().replaceAll("\\[", "\\\\[").replaceAll("\\]", "\\\\]") // Score
+                    + "\"\\}"; // End of element
+        } else {
+            regex = "\\{\\}";
```

Comment:
I don't much like regexes in Java, as they become very hard to read. Did you consider using `\Q no need to escape in here \E` to reduce some of that escaping?

Reply:
What do you mean by `\Q no need to escape in here \E`? Though I do agree that these Java regexes are very ugly, see another example in [optaplanner-persistence-jaxb](https://github.com/kiegroup/optaplanner/blob/master/optaplanner-persistence/optaplanner-persistence-jaxb/src/test/java/org/optaplanner/persistence/jaxb/api/score/AbstractScoreJaxbXmlAdapterTest.java#L108). This PR actually is consistent with that style, though we might as well overhaul all of them in a separate PR. @ge0ffrey WDYT?

Reply:
In a Java regex string, you can use `\Q` to start a sequence in which escaping is not necessary. `\E` ends that sequence.

Reply:
I see, yes that would be a lot cleaner. In any case, we would have to change this for all the other modules, so I'm going to hide behind scope creep for this PR.